### ISSUE TITLE
`function-type`: false negative when function call is given as constant

### ISSUE THREAD

**Issue by jrencz** [Gillespie59/eslint-plugin-angular#407]
Example

```js
angular
  .module('foo', [])
  .constant('foo', Symbol())
```

Triggers:
` 1:1  error  Use anonymous functions instead of named function  angular/function-type`

Which it shouldn't

plugin version: 1.3.1

Possibly related: #376

**Comment by jrencz** [comment]
other example:

```js
angular
  .module('foo', [])
  .constant('foo', Object.freeze({}))
```

**Comment by jrencz** [comment]
another example: IIFE

```js
angular
  .module('foo', [])
  .constant('foo', (() => {})())
```

**Comment by ContemporaryConsultant** [comment]
It seems that still is not fixed for constants with arrays:
angular.module('navigation')
    .constant('NAV_STATES', [ {name: 'Dashboard', state: 'app.dashboard'}]);

This causes the same error as above.
